webapp/app/.server/service/proposal-times.ts

```diff
@@ -0,0 +1,41 @@
+import { PROPOSAL_STATES, type StatusTime } from "@/utils/proposal-states";
+
+export function daysInSeconds(days: number): number {
+  return days * 24 * 3600;
+}
+
+export function calculateStatusPendingDays(
+  status: PROPOSAL_STATES,
+  creationTimestamp: number,
+  guardiansExtendedLegalVeto: boolean,
+  nowInSeconds: number
+): StatusTime | null {
+  if (status === PROPOSAL_STATES.LegalVetoPeriod) {
+    const delta = nowInSeconds - creationTimestamp;
+    const currentDay = Math.ceil(delta / daysInSeconds(1));
+    const totalDays = guardiansExtendedLegalVeto ? 7 : 3;
+
+    return {
+      totalDays: totalDays,
+      currentDay: currentDay,
+    };
+  }
+
+  if (status === PROPOSAL_STATES.Waiting) {
+    const days3 = daysInSeconds(3);
+    const days7 = daysInSeconds(7);
+    const vetoPeriodDuration = guardiansExtendedLegalVeto ? days7 : days3;
```

Comment:
Enshrine this as a const at top of file so if this changes or needs to be refactored it's more obvious
(whether it's 7 days or 3, or even if it needs to be read from somewhere)